### PR TITLE
Mobile paratime selector

### PR DESCRIPTION
**Pull request by lubej**
## Solution

Add mobile version of the paratime selector

- add `react-quick-pinch-zoom` dependency for zooming in/out and pan actions on mobile;
- integrate `react-quick-pinch-zoom` dependency;
- add `use-resize-observer` dependency for implementation of `ResizeObserver`;

## Caviats

- not optimized yet for landscape mode - probably enough for MVP;
- help screen will be done in additional PR;
- mobile version of the search component needs to be modified;

## Links

- https://www.loom.com/share/80b243eff549476584881698f6f2f035

### CODE REVIEW

File: src/app/pages/HomePage/ParaTimeSelector/index.tsx [hunk 1]
```diff
@@ -71,41 +81,95 @@ const ZoomOutBtnFade = styled(Fade)(() => ({
...
 
   const onZoomOutClick = () => {
-    setSelectedGraphEndpoint(GraphEndpoint.CONSENSUS)
+    setSelectedGraphEndpoint(GraphEndpoint.Consensus)
   }
 
+  const onUpdate = ({ x, y, scale }: UpdateAction) => {
+    if (!isMobile) {
+      return
+    }
```
Comment (lukaw3d) [inline src/app/pages/HomePage/ParaTimeSelector/index.tsx:130]: I would name this "onPinchZoom" and remove the `if (!isMobile)`

File: src/app/pages/HomePage/ParaTimeSelector/index.tsx [hunk 2]
```diff
@@ -10,7 +10,12 @@ import { ParaTimeSelectorStep } from './types'
 import { ParaTimeSelectorUtils } from './para-time-selector-utils'
 import { GraphEndpoint } from './Graph/types'
 import Fade from '@mui/material/Fade'
+import useMediaQuery from '@mui/material/useMediaQuery'
+import QuickPinchZoom from 'react-quick-pinch-zoom'
+import PinchZoom, { make3dTransformValue, UpdateAction } from 'react-quick-pinch-zoom'
```
Comment (lukaw3d) [inline src/app/pages/HomePage/ParaTimeSelector/index.tsx:15]: QuickPinchZoom == PinchZoom

File: src/app/pages/HomePage/ParaTimeSelector/index.tsx [hunk 3]
```diff
@@ -71,41 +81,95 @@ const ZoomOutBtnFade = styled(Fade)(() => ({
...
+    setSelectedGraphEndpoint(GraphEndpoint.Consensus)
   }
 
+  const onUpdate = ({ x, y, scale }: UpdateAction) => {
+    if (!isMobile) {
+      return
+    }
+
+    const transformValue = make3dTransformValue({ x, y, scale })
+    quickPinchZoomInnerRef.current?.style.setProperty('transform', transformValue)
```
Comment (lukaw3d) [inline src/app/pages/HomePage/ParaTimeSelector/index.tsx:133]: Having two different systems for transforming two different elements seems completely unmaintainable

File: src/app/pages/HomePage/index.tsx [hunk 4]
```diff
@@ -54,19 +82,21 @@ export const HomePage: FC = () => {
...
         </Box>
       </Content>
-      <FooterStyled>
-        <Footer />
-      </FooterStyled>
+      {!isMobile && (
+        <FooterStyled>
+          <Footer />
+        </FooterStyled>
+      )}
```
[the remark this answers is not in the document]
Comment (lubej) [reply src/app/pages/HomePage/index.tsx:99]: image

The footer on mobile page has the action for showing the info screen. Might look cramped up, if there is additional info in the footer.

File: src/app/pages/HomePage/ParaTimeSelector/index.tsx (hunk 1 above)
Comment (lubej) [reply src/app/pages/HomePage/ParaTimeSelector/index.tsx:130]: Right, this function won't be called on mobile. Good catch 👍

File: src/app/pages/HomePage/index.tsx [hunk 5]
```diff
@@ -1,21 +1,22 @@
...
+const HomepageLayout = styled(Box)(({ theme }) => ({
   position: 'relative',
   display: 'flex',
   flexDirection: 'column',
   width: '100vw',
-  height: '100vh',
-  minHeight: '800px',
+  height: 'fill-available',
   backgroundColor: COLORS.brandDark,
+  overflow: 'hidden',
```
[the remark this answers is not in the document]
Comment (lubej) [reply src/app/pages/HomePage/index.tsx:19]: It has to do with how the ParaTimeSelector is positioned on mobile. It has width: '130vw', to achieve the "overflow" effect. Let me see if I can find some other solution for it.

File: src/app/pages/HomePage/ParaTimeSelector/index.tsx (hunk 3 above)
Comment (lubej) [reply src/app/pages/HomePage/ParaTimeSelector/index.tsx:133]: Right. Let me see if I can use QuickPinchZoom in both cases.

File: src/app/pages/HomePage/ParaTimeSelector/Graph/index.tsx [hunk 6]
```diff
@@ -15,21 +15,22 @@ interface GraphBaseProps {
 
 interface GraphProps extends GraphBaseProps {
   // TODO: Consider moving this to a state management solution
-  selectedGraphEndpoint: GraphEndpoint
-  setSelectedGraphEndpoint: Dispatch<SetStateAction<GraphEndpoint>>
+  selectedGraphEndpoint?: GraphEndpoint
+  setSelectedGraphEndpoint: Dispatch<SetStateAction<GraphEndpoint | undefined>>
```
[the remark this answers is not in the document]
Comment (lubej) [reply src/app/pages/HomePage/ParaTimeSelector/Graph/index.tsx:19]: Right, I was just using the full definition of the `useState`
`function useState<S>(initialState: S | (() => S)): [S, Dispatch<SetStateAction<S>>];`

File: src/app/pages/HomePage/ParaTimeSelector/index.tsx (hunk 3 above)
Comment (lubej) [reply src/app/pages/HomePage/ParaTimeSelector/index.tsx:128]: Resolved, QuickPinchZoom used in both cases - for mobile and desktop.

File: src/app/pages/HomePage/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (lubej) [reply src/app/pages/HomePage/index.tsx:19]: Not sure how to effectively avoid this, do you maybe have any ideas?

[the remark this answers is not in the document]
Comment (lukaw3d) [reply src/app/pages/HomePage/index.tsx:19]: A big improvement would be to use `overflow-x: hidden`

An alternative approximation:
```tsx
const ParaTimeSelectorGlobe = styled(Box)(() => ({
  boxShadow: `
    0px -50px 60px #02fbfdcc,
    0px 0px 60px #02fbfdcc,
    0px 50px 60px #02fbfdcc
  `,
  clipPath: 'circle(72% at 50% 34%)',
```

**Review by lukaw3d — APPROVED** [review]

**Pull request merged**